src/wallet/rpcdump.cpp

```diff
@@ -428,6 +432,10 @@ UniValue importpubkey(const JSONRPCRequest& request)
     if (fRescan && fPruneMode)
         throw JSONRPCError(RPC_WALLET_ERROR, "Rescan is disabled in pruned mode");
 
+    if (fRescan && pwallet->IsScanning()) {
```

Comment:
This is very race-y. You could check check this, have another thread call rescan, and then end up with two parallell rescans.